snapcraft/_options.py

```diff
@@ -0,0 +1,41 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+class _Project:
+
+    @property
+    def use_geoip(self):
+        return self.__use_geoip
+
+    @use_geoip.setter
+    def use_geoip(self, setting):
+        self.__use_geoip = setting
```

Comment:
> Another thing I was contemplating was making the values set immutable and only allow them to be set on class instantiation. Thoughts?

I'm not sure I understand what you mean by "values set" here. You mean the values of this class? e.g. so plugins can't change these settings for the entire project?